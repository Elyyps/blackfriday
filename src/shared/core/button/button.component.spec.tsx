import React from "react";
import { shallow } from "enzyme";
import { Button } from "./";
import IconDefault from "@assets/icons/link.svg";

describe("[Button]", () => {
  it("Should render button with icon right without crashing", () => {
    const renderedComponent = shallow(
      <Button icon={IconDefault} iconPosition="right" title="Primary" variant="primary-default" />
    );
    expect(renderedComponent).toMatchSnapshot();
  });
  it("Should render button with icon left and full width without crashing", () => {
    const renderedComponent = shallow(
      <Button icon={IconDefault} iconPosition="left" title="Primary" variant="primary-default" fullWidth />
    );
    expect(renderedComponent).toMatchSnapshot();
  });
  it("Should render button without icon without crashing", () => {
    const renderedComponent = shallow(<Button title="Primary" variant="primary-default" />);
    expect(renderedComponent).toMatchSnapshot();
  });
  it("Should render button without text (FAB) without crashing", () => {
    const renderedComponent = shallow(<Button icon={IconDefault} iconPosition="left" variant="primary-default" />);
    expect(renderedComponent).toMatchSnapshot();
  });
});
