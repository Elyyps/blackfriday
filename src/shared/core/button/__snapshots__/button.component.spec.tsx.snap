// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[Button] Should render button with icon left and full width without crashing 1`] = `
<Fragment>
  <button
    className="button button--primary-default button--primary-default--icon-filled button--fullWidth"
    name="Primary"
    style={
      Object {
        "height": undefined,
        "width": "100%",
      }
    }
  >
    <span
      className="icon-svg"
    >
      <span
        className="icon-left"
      >
        <IconComponent
          icon="link.svg"
          size="14px"
        />
      </span>
      Primary
    </span>
  </button>
</Fragment>
`;

exports[`[Button] Should render button with icon right without crashing 1`] = `
<Fragment>
  <button
    className="button button--primary-default button--primary-default--icon-filled"
    name="Primary"
    style={
      Object {
        "height": undefined,
        "width": undefined,
      }
    }
  >
    <span
      className="icon-svg"
    >
      Primary
      <span
        className="icon-right"
      >
        <IconComponent
          icon="link.svg"
          size="14px"
        />
      </span>
    </span>
  </button>
</Fragment>
`;

exports[`[Button] Should render button without icon without crashing 1`] = `
<Fragment>
  <button
    className="button button--primary-default button--primary-default--icon-filled"
    name="Primary"
    style={
      Object {
        "height": undefined,
        "width": undefined,
      }
    }
  >
    <span
      className="icon-svg"
    >
      Primary
    </span>
  </button>
</Fragment>
`;

exports[`[Button] Should render button without text (FAB) without crashing 1`] = `
<Fragment>
  <button
    className="button button--primary-default button--FAB button--primary-default--icon-filled"
    style={
      Object {
        "height": undefined,
        "width": undefined,
      }
    }
  >
    <span
      className="icon-svg"
    >
      <span
        className="icon-"
      >
        <IconComponent
          icon="link.svg"
          size="14px"
        />
      </span>
    </span>
  </button>
</Fragment>
`;
